pkg/image/admission/imagepolicy/api/types.go

```diff
@@ -53,13 +55,21 @@ var (
 
 // ImageResolutionPolicyRule describes resolution rules based on resource.
 type ImageResolutionPolicyRule struct {
+	// Policy controls whether resolution will happen if the rule doesn't match local names. This value
+	// overrides the global image policy for all target resources.
+	Policy ImageResolutionType
 	// TargetResource is the identified group and resource. If Resource is *, this rule will apply
 	// to all resources in that group.
 	TargetResource metav1.GroupResource
 	// LocalNames will allow single segment names to be interpreted as namespace local image
 	// stream tags, but only if the target image stream tag has the "resolveLocalNames" field
 	// set.
 	LocalNames bool
+	// Type controls whether resolution will happen if the rule doesn't match local names. The default
+	// value is DoNotAttempt. This overrides the global image policy for a matching resource. If there
+	// are multiple rules matching a resource, any rule that requests resolution or rewrite will
+	// result in rewrite.
+	Type ImageResolutionType
```

Comment:
@smarterclayton isn't this duplicate to `ImageResolutionType`? seems like rebase going wrong